services/consensusalgo/benchmarkconsensus/test/harness.go

```diff
@@ -71,3 +76,82 @@ func (h *harness) createService(ctx context.Context) {
 		h.config,
 	)
 }
+
+func nonLeaderPublicKey() primitives.Ed25519Pkey {
+	return []byte{0x99}
+}
+
+func (h *harness) expectHandlerRegistrations() {
+	h.gossip.Reset().When("RegisterBenchmarkConsensusHandler", mock.Any).Return().Times(1)
+	h.blockStorage.Reset().When("RegisterConsensusBlocksHandler", mock.Any).Return().Times(1)
+}
+
+func (h *harness) verifyHandlerRegistrations(t *testing.T) {
+	ok, err := h.gossip.Verify()
+	if !ok {
+		t.Fatal("Did not register with Gossip:", err)
+	}
+	ok, err = h.blockStorage.Verify()
+	if !ok {
+		t.Fatal("Did not register with BlockStorage:", err)
+	}
+}
+
+func (h *harness) expectBlockCreation() {
+	h.consensusContext.Reset().When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).AtLeast(1)
+}
+
+func (h *harness) verifyBlockCreation(t *testing.T) {
+	err := test.EventuallyVerify(h.consensusContext)
+	if err != nil {
+		t.Fatal("Did not create block with ConsensusContext:", err)
+	}
+}
+
+func (h *harness) expectNoBlockCreation() {
+	h.consensusContext.Reset().When("RequestNewTransactionsBlock", mock.Any).Return(nil, nil).Times(0)
+}
+
+func (h *harness) verifyNoBlockCreation(t *testing.T) {
+	err := test.ConsistentlyVerify(h.consensusContext)
+	if err != nil {
+		t.Fatal("Did create block with ConsensusContext:", err)
+	}
+}
+
+func (h *harness) receiveCommit(blockPair *protocol.BlockPairContainer) {
+	h.service.HandleBenchmarkConsensusCommit(&gossiptopics.BenchmarkConsensusCommitInput{
+		Message: &gossipmessages.BenchmarkConsensusCommitMessage{
+			BlockPair: blockPair,
+		},
+	})
+}
+
+func (h *harness) expectIgnoreCommit() {
+	h.blockStorage.Reset().When("CommitBlock", mock.Any).Return(nil, nil).Times(0)
```

Comment:
why do you need to call Reset()? this is usually a smell...

Reply:
Some of these mocks are required for construction (like the register). This means the harness always sets them just so it won't panic. If the test explicitly wants to set them in order to verify how they were called I preferred to reset so they won't accumulate. I'm basically replacing the default behavior of the harness here